### ISSUE TITLE
BUG: v1.45-031 Dropbox-API-Arg": unknown field 'cursor'

### ISSUE THREAD

**Issue by ajkis** [ncw/rclone#2829]
Version: rclone v1.45-031-ge7684b7e-beta
```
2018/12/10 11:44:12 ERROR : series/God Friended Me/Season 01/God Friended Me - S01E10 - WEBDL-1080p.mkv: Failed to copy: upload failed: Error in call to API function "files/upload_session/append": HTTP header "Dropbox-API-Arg": unknown field 'cursor'
2018/12/10 11:44:12 ERROR : series/God Friended Me/Season 01/God Friended Me - S01E10 - WEBDL-1080p.mkv: Not deleting source as copy failed: upload failed: Error in call to API function "files/upload_session/append": HTTP header "Dropbox-API-Arg": unknown field 'cursor'
2018/12/10 11:44:21 ERROR : series/Keeping Up with the Kardashians/Season 15/Keeping Up with the Kardashians - S15E16 - WEBDL-1080p.mkv: Failed to copy: upload failed: Error in call to API function "files/upload_session/append": HTTP header "Dropbox-API-Arg": unknown field 'cursor'
2018/12/10 11:44:21 ERROR : series/Keeping Up with the Kardashians/Season 15/Keeping Up with the Kardashians - S15E16 - WEBDL-1080p.mkv: Not deleting source as copy failed: upload failed: Error in call to API function "files/upload_session/append": HTTP header "Dropbox-API-Arg": unknown field 'cursor'
2018/12/10 11:44:23 ERROR : series/Madam Secretary/Season 05/Madam Secretary - S05E09 - WEBDL-1080p.mkv: Failed to copy: upload failed: Error in call to API function "files/upload_session/append": HTTP header "Dropbox-API-Arg": unknown field 'cursor'
2018/12/10 11:44:23 ERROR : series/Madam Secretary/Season 05/Madam Secretary - S05E09 - WEBDL-1080p.mkv: Not deleting source as copy failed: upload failed: Error in call to API function "files/upload_session/append": HTTP header "Dropbox-API-Arg": unknown field 'cursor'
2018/12/10 11:44:30 ERROR : series/The Flash (2014)/Season 05/The Flash (2014) - S05E09 - WEBDL-1080p.mkv: Failed to copy: upload failed: Error in call to API function "files/upload_session/append": HTTP header "Dropbox-API-Arg": unknown field 'cursor'
2018/12/10 11:44:30 ERROR : series/The Flash (2014)/Season 05/The Flash (2014) - S05E09 - WEBDL-1080p.mkv: Not deleting source as copy failed: upload failed: Error in call to API function "files/upload_session/append": HTTP header "Dropbox-API-Arg": unknown field 'cursor'
```

**Comment by ncw** [comment]
Err, yes it looks like 4138d5aa751bf9c31e8aee3307a513c9b0d49d5a or 8fb707e16dab63609bf29b0eac65cf6395af8942 has broken the dropbox backend.

The integration tests are unhappy too: https://pub.rclone.org/integration-tests/current/

@mathieucarbou have you got time to look at this?

**Comment by mathieucarbou** [comment]
@ncw : This is not 8fb707e16dab63609bf29b0eac65cf6395af8942.

But this might be 4138d5aa751bf9c31e8aee3307a513c9b0d49d5a (the dropbox sdk upgrade). 

@diwakergupta changed a lot of things and upgraded to a major version both SDK and api specs.
Look at the commit history here: https://github.com/dropbox/dropbox-sdk-go-unofficial/commits/master).

My original patch was just about updating some specific files inside the the vendor version you were pointing to but you preferred to link to the next Dropbox version and tag (which is, imo, better also). But it contains risks obviously of bringing new changes. I thought you would be OK with and you had plan on looking / reviewing / testing the dropbox integration carefully. 

Sadly I do not have time to look at all the possible changes they did and that could break thinks

@ajkis : to help us, are you able to re-run your command with `--log-level DEBUG` and give us more details about the error plus all the the rclone command and command-line args you are using ? (without any username / password)

Thanks!

**Comment by mathieucarbou** [comment]
@ncw : BTW, looking at my PR that you have merged: https://github.com/ncw/rclone/pull/2622, the checks were successful...

![image](https://user-images.githubusercontent.com/61346/49741066-982f5200-fc63-11e8-93a3-0564512b9346.png)

Are you sure it is these commits then ?
Are the PR builds also running the integration tests (which they should at least) ?

**Comment by ncw** [comment]
> Are the PR builds also running the integration tests (which they should at least) ?

No they don't - they are too expensive (time and money wise) to do that.



> I strongly advice though to fix the integration tests... Because right now, knowing that Dropbox Team has upgraded its SDK to 5 with some changes, anyway rclone has no choice but to adapt to it.

Yes that is what I usually do.  Have you got time to help with this?

**Comment by mathieucarbou** [comment]
> Yes that is what I usually do. Have you got time to help with this?

Yes and no... I only have little time sporadically until January (moving in progress). I had a quick look yesterday and I do not understand the complain about the cursor being in the http header during a chunk upload. The cursor field is part of the dropbox api. I do not have enough Go knowledge also to be able to find a potential change to apply in the rclone code, or a potential bug in dropbox api...

So in conclusion, my help will be limited and not as valuable as you might expect.

**Comment by ncw** [comment]
Things are a bit hectic right now for me too, but I'm hoping to have a bit of time over the holiday season.

Thanks for your help @mathieucarbou :-)

**Comment by kernalkue** [comment]
I'm having big problems with this bug. Is there a work around?

**Comment by ncw** [comment]
Sorry everyone, I have a bit of a backlog!

I've figured out the problem with the dropbox SDK: https://github.com/dropbox/dropbox-sdk-go-unofficial/issues/49 and I've sent a PR to fix https://github.com/dropbox/dropbox-sdk-go-unofficial/issues/50

Here is a version of rclone with the SDK patch I submitted in.  It now passes all the integration tests so I'll merge it after it has run through the CI.

@ajkis and @kernalkue can you give it a test to see if it fixes your problem?

https://beta.rclone.org/branch/v1.45-038-g3c624c6f-fix-2829-dropbox-sdk-beta/ (uploaded in 15-30 mins)

**Comment by ncw** [comment]
The PR has been applied to the dropbox repo and I've pulled it in and checked it isOK.

So the fix will be in in [the latest beta](https://beta.rclone.org) in 15-30 mins.

**Comment by ajkis** [comment]
@mathieucarbou @ncw   sorry I was away over Christmas, just tested the latest beta and so far so good.

**Comment by mathieucarbou** [comment]
@ncw : I can confirm that sync is working plus the Retry-After support 👍

**Comment by ncw** [comment]
> I can confirm that sync is working plus the Retry-After support

Thanks for testing :-)

> So... finally... It was a bug within the SDK ?

Yes it was... The SDK got re-organised and all the `_v2` dropped off the API endpoints!

I think this issue is done now so I'll close it :-)
